tools/js-sdk-release-tools/packages/typescript-codegen-breaking-change-detector/src/tests/patch-detection.test.ts

```diff
@@ -1,183 +1,333 @@
+      expect(breakingPairs.length).toBe(0);
+    });
+
+    test('remove constructors', async () => {
+      const baselineApiView = `
+      class RemoveClassConstructor {
+        constructor(remove: string) {}
+        constructor(p1: string, p2: string) {}
+      }`;
+      const currentApiView = `
+      class RemoveClassConstructor {
+      }`;
+
+      const astContext = createTestAstContext(baselineApiView, currentApiView);
+      const breakingPairs = patchClass('RemoveClassConstructor', astContext, AssignDirection.CurrentToBaseline);
+      expect(breakingPairs.length).toBe(2);
+      expect(breakingPairs[0].assignDirection).toBe(AssignDirection.CurrentToBaseline);
+      expect(breakingPairs[0].location).toBe(DiffLocation.Signature);
+      expect(breakingPairs[0].reasons).toBe(DiffReasons.Removed);
+      expect(breakingPairs[0].target?.name).toBe('constructor(remove: string) {}');
+      expect(breakingPairs[1].assignDirection).toBe(AssignDirection.CurrentToBaseline);
+      expect(breakingPairs[1].location).toBe(DiffLocation.Signature);
+      expect(breakingPairs[1].reasons).toBe(DiffReasons.Removed);
+      expect(breakingPairs[1].target?.name).toBe('constructor(p1: string, p2: string) {}');
+    });
+
+    test('change type of constructor\'s parameter', async () => {
+      const baselineApiView = `
+      class TestClass {
+        constructor(p1: string, p2: string) {}
+      }`;
+      const currentApiView = `
+      class TestClass {
+        constructor(p2: string, p2: number) {}
+      }`;
+
+      const astContext = createTestAstContext(baselineApiView, currentApiView);
+      const breakingPairs = patchClass('TestClass', astContext, AssignDirection.CurrentToBaseline);
       expect(breakingPairs.length).toBe(1);
       expect(breakingPairs[0].assignDirection).toBe(AssignDirection.CurrentToBaseline);
-      expect(breakingPairs[0].location).toBe(DiffLocation.TypeAlias);
-      expect(breakingPairs[0].reasons).toBe(DiffReasons.TypeChanged);
-      expect(breakingPairs[0].target?.name).toBe('typesChange');
+      expect(breakingPairs[0].location).toBe(DiffLocation.Signature);
+      expect(breakingPairs[0].reasons).toBe(DiffReasons.Removed);
+      expect(breakingPairs[0].target?.name).toBe('constructor(p1: string, p2: string) {}');
+    });
 
-      breakingPairs = patchTypeAlias('typesRemove', astContext, AssignDirection.CurrentToBaseline);
+    test('change name of constructor\'s parameter', async () => {
+      const baselineApiView = `
+      class TestClass {
+        constructor(p1: string, p2: string) {}
+      }`;
+      const currentApiView = `
+      class TestClass {
+        constructor(p2: string, p3: string) {}
+      }`;
+
+      const astContext = createTestAstContext(baselineApiView, currentApiView);
+      const breakingPairs = patchClass('TestClass', astContext, AssignDirection.CurrentToBaseline);
```

Comment:
`breakingPairs` is this for breaking? i think this is just a diff pair? do we know if this diff change is a breaking?

Changing param name of constructor is not a breaking for me.

Reply:
Got it, let me fix in next PR. -- This PR is 3 months ago from now, and it's the only issue, that no impact for current detection. No need to change it and review again. Please ping me if you have concerns